tutorials/dataframe/tdf012_InspectAnalysis.C

```diff
@@ -0,0 +1,119 @@
+   auto h = tdf.Histo1D<double>({"browserHisto", "", 100, -2., 2.}, "x");
+
+   // ## Use the callback mechanism to draw the histogram on a TBrowser while it is being filled
+   // So far we have registered a column "x" to a data-frame with `nEvents` events and we registered the filling of a
+   // histogram with the values of column "x".
+   // In the following we will register three functions for execution during the event-loop:
+   // - one is to be executed once just before the loop and adds a partially-filled histogram to a TBrowser
+   // - the next is executed every 50 events and draws the partial histogram on the TBrowser's TPad
+   // - another callback is responsible of updating a simple progress bar from multiple threads
+
+   // First off we create a TBrowser that contains a "TDFResults" directory
+   TMemFile tdfDirectory("TDFResults", "RECREATE");
+   TBrowser browser("b", &tdfDirectory);
+   // The global pad should now be set to the TBrowser's canvas, let's store its value in a local variable
+   auto browserPad = gPad;
+
+   // A useful feature of `TResultProxy` is its `OnPartialResult` method: it allows us to register a callback that is
+   // executed once per specified number of events during the event-loop, on "partial" versions of the result objects
+   // contained in the `TResultProxy`. In this case, the partial result is going to be a histogram filled with an
+   // increasing number of events.
+   // Instead of requesting the callback to be executed every N entries, this time we use the special value `kOnce` to
+   // request that it is executed once right before starting the event-loop.
+   // The callback is a C++11 lambda that registers the partial result object in `tdfDirectory`.
+   h.OnPartialResult(h.kOnce, [&tdfDirectory](TH1D &h_) { tdfDirectory.Add(&h_); });
+   // Note that we called `OnPartialResult` with a dot, `.`, since this is a method of `TResultProxy` itself.
+   // We do not want to call `OnPartialResult` on the pointee histogram!)
+
+   // Multiple callbacks can be registered on the same `TResultProxy` (they are executed one after the other in the
+   // same order as they were registered). We now request that the partial result is drawn and the TBrowser's TPad is
+   // updated every 50 events.
+   h.OnPartialResult(50, [&browserPad](TH1D &hist) {
+      browserPad->cd();
+      hist.Draw();
+      browserPad->Update();
+      // This call tells ROOT to process all pending GUI events
+      // It allows users to use the TBrowser as usual while the event-loop is running
+      gSystem->ProcessEvents();
+   });
+
+   // Finally, we would like to print a progress bar on the terminal to show how the event-loop is progressing.
+   // To take into account _all_ events we use `OnPartialResultSlot`: when Implicit Multi-Threading is enabled, in fact,
+   // `OnPartialResult` invokes the callback only in one of the worker threads, and always returns that worker threads'
```

Comment:
This sounds like it is executed on an arbitrary and potential different thread each time, is that the intent?

Reply:
yes, as far as the user is concerned, that's correct.

I cannot (performantly) pin the execution of the callback to a specific thread id. I can pin it to a specific "slot" number (the 0-th slot) but slots can be taken up by different threads over the course of an event loop.

Users should not rely on the callback being always executed by the same thread. This is also explicitly stated at TResultProxy.hxx:213, in the documentation of `OnPartialResult`.